src/components/Form/Checkbox.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { Component, ReactNode } from 'react';
+import classnames from 'classnames';
+import { HtmlInput, HtmlLabel, HtmlSpan, HtmlDiv } from '../../reset';
+
+import { idGenerator } from '../../utils/uuid';
+
+const baseClassName = 'fi-checkbox';
+
+const checkboxBaseClassNames = {
+  disabled: `${baseClassName}--disabled`,
+  container: `${baseClassName}_container`,
+  input: `${baseClassName}_input`,
+  label: `${baseClassName}_label`,
+  statusText: `${baseClassName}_status`,
+  hintText: `${baseClassName}_hintText`,
+};
+
+type CheckboxVariant = 'small' | 'large';
+type CheckboxStatus = 'default' | 'error' | 'disabled';
+
+export interface CheckboxInputProps {
+  /** State of input checkbox */
+  checked?: boolean;
+  /** Custom classname for the container */
+  className?: string;
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+}
+
+export interface CheckboxProps {
+  /** Controlled checked-state - user actions use onClick to change  */
+  checked?: boolean;
+  /** Default status of Checkbox when not using controlled 'checked' state
+   * @default false
+   */
+  defaultChecked?: boolean;
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Disable Checkbox. Value won't be included when submitting */
+  disabled?: boolean;
+  /** Event handler to execute when clicked */
+  onClick?: ({ checkboxState }: { checkboxState: boolean }) => void;
+  /**
+   * Label element content
+   */
+  children?: ReactNode;
+  /**
+   * 'small' | 'large'
+   * @default small
+   */
+  variant?: CheckboxVariant;
+  /**
+   * 'default' | 'error' | 'disabled'
+   * @default default
+   */
+  status?: CheckboxStatus;
```

Comment:
Would it be better to have the disabled prop removed and that behaviour controlled by the status? Or would that be inconsistent with checkbox JSX props? Now one would have to do <Checkbox disabled status="disabled" />

Reply:
As discussed, it is also possible to check if disabled exists and dismiss other status props if so. Allowing e.g. disabled and checked would require styles to be defined for designs etc.

Reply:
I'd say that's something we can leave for the users of the component. For example, in some contexts you might want to show a disabled checkbox with an error message if an action is required by a user to enable the checkbox. The disabled+checked variant can be found in the styleguide, so I'd leave that as it is.

Reply:
I think the behavior of the API should be deterministic and easy to understand. Are these equivalent: `<Checkbox status='disabled'>` and `<Checkbox disabled>`? What happens if I have this `<Checkbox disabled status='error'>`? 

I propose the following:
1) Remove disabled from CheckboxStatus
2) Either dismiss the error style if provided with disabled, or define the style so that it works
3) If we support both disabled and error at the same time, ensure that there is matching design and we follow that.

Reply:
The disabled prop is present in other components as well. As discussed, to keep the API's consistent let's keep it that way and just make sure we disable other status related styles when disabled is present.

Reply:
Other status styles are now turned off when the component is disabled.